pkg/portForward/portForward.go

```diff
@@ -58,7 +66,6 @@ func (o *PFClient) StartPortForwarding(
 	o.StopPortForwarding()
 
 	o.stopChan = make(chan struct{}, 1)
```

Comment:
Why is this a buffered channel?

Reply:
Using a non-buffered one, writing into this channel will block until the someone reads the channel.

Usin a buffered one, the first write into the channel won't block, even if no one reads it.

This way, we can write on the channel and continue, even if the `forward` function is not yet started and so wion't read the channel. 

That's also why I'm recreating a fresh new channel after, so if the channel has not been read, we won't be blocked during the second write.

Reply:
But of which size? If you create it of size `n`, it will happen once in a while that it will be written `n+1` times.

Particularly, I tested the changes by flooding the app with the following command, while modifying source code so the pod restarts:

```
while : ; do curl localhost:40001; done
```

With a buffered channel of size 100, I'm pretty sure I would have filled the buffer

Reply:
For channels sizing, see this recommendation (size of 0 or 1) from the coding conventions doc: https://github.com/redhat-developer/odo/wiki/Dev:-Coding-Conventions#channel-size-is-one-or-none